src/Annotations/Serializer/AnnotationNormalizer.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace eLife\Annotations\Serializer;
+
+use eLife\ApiSdk\ApiSdk;
+use eLife\HypothesisClient\Model\Annotation;
+use League\CommonMark\Block\Element;
+use League\CommonMark\DocParser;
+use League\CommonMark\ElementRendererInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerAwareTrait;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+final class AnnotationNormalizer implements NormalizerInterface, NormalizerAwareInterface
+{
+    const CANNOT_RENDER_CONTENT_COPY = 'NOTE: It is not possible to display this content.';
+
+    use NormalizerAwareTrait;
+
+    private $docParser;
+    private $htmlRenderer;
+    private $logger;
+
+    public function __construct(DocParser $docParser, ElementRendererInterface $htmlRenderer, LoggerInterface $logger)
+    {
+        $this->docParser = $docParser;
+        $this->htmlRenderer = $htmlRenderer;
+
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param Annotation $object
+     */
+    public function normalize($object, $format = null, array $context = []) : array
+    {
+        $content = $object->getText() ? $this->processText($object->getText()) : null;
+
+        $data = array_filter([
+            'id' => $object->getId(),
+            'access' => ($object->getPermissions()->getRead() === Annotation::PUBLIC_GROUP) ? 'public' : 'restricted',
+            'content' => $content,
+            'parents' => $object->getReferences(),
+            'created' => $object->getCreatedDate()->format(ApiSdk::DATE_FORMAT),
+            'updated' => $object->getUpdatedDate()->format(ApiSdk::DATE_FORMAT),
+            'document' => [
+                'title' => $object->getDocument()->getTitle(),
+                'uri' => $object->getUri(),
+            ],
+        ]) + ['parents' => []];
+        if ($data['created'] === $data['updated']) {
+            unset($data['updated']);
+        }
+        if ($object->getTarget()->getSelector()) {
+            $data['highlight'] = $object->getTarget()->getSelector()->getTextQuote()->getExact();
+        }
+        if (empty($data['highlight']) && empty($data['content'])) {
+            $this->logger->warning(sprintf('Annotation detected without highligt or content (ID: %s)', $data['id']), ['annotation' => $data]);
```

Comment:
typo in `highligt`

Reply:
can probably be a level higher than `warning`? Seems a bad situation

Reply:
We cannot absolutely rule it out. If we detect them I would like to be notified.